### ISSUE TITLE
cannot build on FreeBSD

### ISSUE THREAD

**Issue by fluca1978** [tada/pljava#151]
I'm trying to install pljava on FreeBSD 11.1 against PostgreSQL 9.6.8 (packaged).
Here is the setup:

```
% uname -v                                                                   
FreeBSD 11.1-RELEASE #0 r321309: Fri Jul 21 02:08:28 UTC 2017     root@releng2.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC 

% gcc --version  
gcc (FreeBSD Ports Collection) 8.0.1 20180325 (experimental)
% g++ --version                                                                         
g++ (FreeBSD Ports Collection) 8.0.1 20180325 (experimental)
% javac -version                                                                                              
javac 1.8.0_162

```

Package `openssl-devel-1.1.0h_1` installed.

Here there's the execution:

```
% mvn -Dnar.cores=1 -X -Pwnosign   clean install
...
[DEBUG] Execute:Java13CommandLauncher: Executing 'gcc' with arguments:
'-Wno-sign-conversion'
'-Wall'
'-Wno-long-long'
'-Wpointer-arith'
'-Wconversion'
'-fPIC'
'-c'
'-fPIC'
'-DPLJAVA_SO_VERSION=1.6.0-SNAPSHOT'
'-DFreeBSD'
'-DGNU_GCC'
'-I/usr/home/luca/pljava/pljava-so/target/nar/javah-include'
'-I/usr/local/openjdk8/include'
'-I/usr/local/openjdk8/include/freebsd'
'-I/usr/local/include'
'-I/usr/local/include/postgresql/server'
'-I/usr/home/luca/pljava/pljava-so/src/main/include'
'-I/usr/home/luca/pljava/pljava-so/target/nar/javah-include'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/AclId.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Any.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Array.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Backend.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/BigDecimal.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Boolean.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Byte.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Coerce.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Composite.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Date.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Double.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/ErrorData.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Exception.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/ExecutionPlan.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Float.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Function.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/HashMap.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/HeapTupleHeader.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/InstallHelper.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Integer.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Invocation.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Iterator.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/JNICalls.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/LargeObject.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Long.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Oid.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/PgObject.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/PgSavepoint.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Portal.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Relation.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SPI.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLInputFromChunk.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLInputFromTuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLOutputToChunk.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLOutputToTuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Session.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Short.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/String.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Time.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Timestamp.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TriggerData.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Tuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TupleDesc.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TupleTable.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/UDT.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Void.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/byte_array.c'

The ' characters around the executable and arguments are
not part of the command.
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Portal.c: In function 'Java_org_postgresql_pljava_internal_Portal__1isPosOverflow':
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Portal.c:338:42: error: 'struct PortalData' has no member named 'posOverflow'
[ERROR]    result = (jboolean)((Portal)p2l.ptrVal)->posOverflow;
[ERROR]                                           ^~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c: In function 'subXactCB':
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c:26:2: warning: enumeration value 'SUBXACT_EVENT_PRE_COMMIT_SUB' not handled in switch [-Wswitch]
[ERROR]   switch(event)
[ERROR]   ^~~~~~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/TupleDesc.c: In function 'Java_org_postgresql_pljava_internal_TupleDesc__1formTuple':
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/type/TupleDesc.c:239:11: warning: implicit declaration of function 'heap_formtuple'; did you mean 'heap_form_tuple'? [-Wimplicit-function-declaration]
[ERROR]    tuple = heap_formtuple(self, values, nulls);
[ERROR]            ^~~~~~~~~~~~~~
[ERROR]            heap_form_tuple
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/type/TupleDesc.c:239:9: warning: assignment to 'HeapTuple' {aka 'struct HeapTupleData *'} from 'int' makes pointer from integer without a cast [-Wint-conversion]
[ERROR]    tuple = heap_formtuple(self, values, nulls);
[ERROR]          ^
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c: In function 'xactCB':
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_COMMIT' not handled in switch [-Wswitch]
[ERROR]   switch(event)
[ERROR]   ^~~~~~
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_ABORT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PRE_PREPARE' not handled in switch [-Wswitch]
[INFO] 51 files were compiled.                                      
[INFO] 51 files were compiled.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] PostgreSQL PL/Java ................................. SUCCESS [  2.702 s]
[INFO] PL/Java API ........................................ SUCCESS [  7.963 s]
[INFO] PL/Java backend Java code .......................... SUCCESS [  3.482 s]
[INFO] PL/Java backend native code ........................ FAILURE [ 21.676 s]
[INFO] PL/Java Ant tasks .................................. SKIPPED
[INFO] PL/Java examples ................................... SKIPPED
[INFO] PL/Java packaging .................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 37.399 s
[INFO] Finished at: 2018-06-11T12:43:08+02:00
[INFO] Final Memory: 32M/78M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.2.3:nar-compile (default-nar-compile) on project pljava-so: NAR: Compile failed: gcc failed with return code 1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.2.3:nar-compile (default-nar-compile) on project pljava-so: NAR: Compile failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: NAR: Compile failed
    at com.github.maven_nar.NarCompileMojo.createLibrary (NarCompileMojo.java:452)
    at com.github.maven_nar.NarCompileMojo.narExecute (NarCompileMojo.java:122)
    at com.github.maven_nar.AbstractNarMojo.execute (AbstractNarMojo.java:302)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.tools.ant.BuildException: gcc failed with return code 1
    at com.github.maven_nar.cpptasks.compiler.CommandLineCompiler.compile (CommandLineCompiler.java:222)
    at com.github.maven_nar.cpptasks.compiler.CommandLineCompilerConfiguration.compile (CommandLineCompilerConfiguration.java:148)
    at com.github.maven_nar.cpptasks.CCTask$Core.run (CCTask.java:946)
```

**Comment by jcflack** [comment]
Hello,
Have you tried one of the [released versions][rels]? 1.5.1-BETA1 is the most recent, and added support for PG 9.6 and 10.

It looks as if you were trying to build the 1.6.0-SNAPSHOT, which is on the development branch, and will eventually become the next major release, but at the moment has not had the latest additions on the 1.5 stable branch merged into it yet.

This happens often enough that I am thinking of editing the `pom.xml` on the development branch to print a warning message. :)

Regards,
-Chap

[rels]: https://github.com/tada/pljava/releases

**Comment by fluca1978** [comment]
Good guess, 

```
% git checkout refs/tags/V1_5_1b1 && mvn clean install
```

worked.
Please note that 1.6 compiled againsta 10.3.1 on the very same machine.

Thanks.
